fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/ContainmentIndexImpl.java

```diff
@@ -775,18 +797,19 @@ public boolean resourceExists(@Nonnull final Transaction tx, final FedoraId fedo
             // Root always exists.
             return true;
         }
-        final MapSqlParameterSource parameterSource = new MapSqlParameterSource();
-        parameterSource.addValue("child", resourceId);
-        final String queryToUse;
         if (tx.isOpenLongRunning()) {
-            queryToUse = includeDeleted ? RESOURCE_OR_TOMBSTONE_EXISTS_IN_TRANSACTION :
+            final var queryToUse = includeDeleted ? RESOURCE_OR_TOMBSTONE_EXISTS_IN_TRANSACTION :
                     RESOURCE_EXISTS_IN_TRANSACTION;
-            parameterSource.addValue("transactionId", tx.getId());
+            return !jdbcTemplate.queryForList(queryToUse,
+                    Map.of("child", resourceId, "transactionId", tx.getId()), String.class).isEmpty();
+        } else if (includeDeleted) {
+            return !jdbcTemplate.queryForList(RESOURCE_OR_TOMBSTONE_EXISTS,
```

Comment:
I understand that you don't want to load the cache here, but what do you think about checking the cache for a hit before doing this query? If there's a hit, you can return, and if it's a miss then you do the query without putting the result in the cache.